src/ImageSharp/Processing/Processors/Transforms/Linear/AutoOrientProcessor{TPixel}.cs

```diff
@@ -81,32 +81,32 @@ protected override void OnFrameApply(ImageFrame<TPixel> sourceBase)
         /// Returns the current EXIF orientation
         /// </summary>
         /// <param name="source">The image to auto rotate.</param>
-        /// <returns>The <see cref="OrientationMode"/></returns>
-        private static OrientationMode GetExifOrientation(Image<TPixel> source)
+        /// <returns>The <see cref="ushort"/></returns>
+        private static ushort GetExifOrientation(Image<TPixel> source)
         {
             if (source.Metadata.ExifProfile is null)
             {
-                return OrientationMode.Unknown;
+                return ExifOrientationMode.Unknown;
             }
 
             IExifValue<ushort> value = source.Metadata.ExifProfile.GetValue(ExifTag.Orientation);
             if (value is null)
             {
-                return OrientationMode.Unknown;
+                return ExifOrientationMode.Unknown;
             }
 
-            OrientationMode orientation;
+            ushort orientation;
             if (value.DataType == ExifDataType.Short)
             {
-                orientation = (OrientationMode)value.Value;
+                orientation = value.Value;
             }
             else
             {
-                orientation = (OrientationMode)Convert.ToUInt16(value.Value);
+                orientation = Convert.ToUInt16(value.Value);
                 source.Metadata.ExifProfile.RemoveValue(ExifTag.Orientation);
```

Comment:
This line should be removed, because the value is always set below:
```suggestion
```

Reply:
We remove to ensure that any invalid values are gone. We've seen in the wild some odd implementations of this property. `SetValue` actually calls `TrySetValue` on the value which can return false.